qutebrowser/config/configfiles.py

```diff
@@ -232,8 +254,17 @@ def read_config_py(filename=None):
             "Unhandled exception",
             exception=e, traceback=traceback.format_exc()))
 
-    api.finalize()
-    return api
+
+def _pre_config_save():
```

Comment:
I think this could look much nicer when you make it a context manager:

```python
import contextlib

...

@contextlib.contextmanager
def add_to_sys_path(path):
    # save / change
    yield
    # restore
```

Then:

```python
with add_to_sys_path(config_dir):
    # do config stuff
```

It nicely separates things (even setting sys.path) from running the config, it has setup/cleanup nicer together, and it automatically takes care of doing the cleanup on exceptions as well.